source/common/http/http1/codec_impl.cc

```diff
@@ -575,7 +594,10 @@ StreamEncoder& ClientConnectionImpl::newStream(StreamDecoder& response_decoder)
   if (resetStreamCalled()) {
     throw CodecClientException("cannot create new streams after calling reset");
   }
-
+  // read-enable any underlying connection disabled by flow control on the prior request.
+  while (!connection_.readEnabled()) {
```

Comment:
Hmm. Yeah now I'm changing my mind. It is better to keep things all together. If you prefer here maybe just comment about why done here vs. connection pool or elsewhere and some more color in comment of what you wrote above?